src/Npgsql/NpgsqlConnection.cs

```diff
@@ -1307,6 +1307,8 @@ public NpgsqlConnection CloneWith(string connectionString)
             var csb = new NpgsqlConnectionStringBuilder(connectionString);
             if (csb.Password == null && Password != null)
                 csb.Password = Password;
+            if (!Settings.PersistSecurityInfo && csb.PersistSecurityInfo)
```

Comment:
Let's swap conditions so it will look like the previous `if`:
```suggestion
            if (csb.PersistSecurityInfo && !Settings.PersistSecurityInfo)
```